client/container.go

```diff
@@ -46,14 +49,38 @@ func (client *APIClient) ContainerStart(name, detachKeys string) error {
 	return err
 }
 
-// ContainerStop stops a container
+// ContainerStop stops a container.
 func (client *APIClient) ContainerStop(name string) error {
 	resp, err := client.post("/containers/"+name+"/stop", nil, nil)
 	ensureCloseReader(resp)
 
 	return err
 }
 
+// ContainerRemove removes a container.
+func (client *APIClient) ContainerRemove(name string, force bool) error {
+	q := url.Values{}
+	if force {
+		q.Set("force", "true")
+	}
+
+	resp, err := client.delete("/containers/"+name, q)
+	if err != nil {
+		return err
+	}
+	defer ensureCloseReader(resp)
+
+	if resp.StatusCode != http.StatusNoContent {
+		body, err := ioutil.ReadAll(resp.Body)
```

Comment:
Is there any necessary for us to add a new struct in swagger.yml, for exmaple `ContainerRmResponse` or something like that? And maybe this will make API data transferring more structed.

Reply:
if remove success, there is no content in body as same as docker. Only ocurring error, the body contain the error message.

Reply:
if string(body) is empty string, this function will return an error. Is it reasonable?
Actually if  `resp.StatusCode != http.StatusNoContent` returns true, it means this is a non-2xx response. Then the error message will not be in `resp`, I think it is in `err` of `resp, err := client.delete("/images/"+name, q)`. This is guaranteed by HTTP framework. So I am wondering if this block is reasonable.  @HusterWan @skoo87

Reply:
Not as you said, if resp.StatusCode != http.StatusNoContent returns true, it means pouchd returns non-204 response. pouchd server remove container success, will returns 204 status code. if not 204, that is remove container failed.  you should look at the server api code.  the action is same as docker, it's reasonable.

Reply:
I am afraid I was wrong. I search moby's client code, in its cli, when requesting, he will do like this:
```
func (cli *Client) sendRequest(ctx context.Context, method, path string, query url.Values, body io.Reader, headers headers) (serverResponse, error) {
	req, err := cli.buildRequest(method, cli.getAPIPath(path, query), body, headers)
	if err != nil {
		return serverResponse{}, err
	}
	resp, err := cli.doRequest(ctx, req)
	if err != nil {
		return resp, err
	}
        //  !!!!!!!! this part is important, transfer error in body to the second return parameter of func
	if err := cli.checkResponseErr(resp); err != nil {
		return resp, err
	}
	return resp, nil
}
```

Reply:
Yes, you are right. I realized it's a problem after I answer you yesterday.